src/models/network/NetworkAppController.js

```diff
@@ -160,46 +145,35 @@ export default class NetworkAppController extends NetworkBaseController {
   }
 
   async linkStdlib() {
-    if (!this.localController.hasStdlib()) {
+    if (!this.packageFile.hasStdlib()) {
       await this.app.setStdlib();
-      delete this.networkPackage['stdlib'];
-      return;
+      return this.networkFile.unsetStdlib()
     }
 
-    const networkStdlib = this.networkPackage.stdlib;
-    const hasNetworkStdlib = !_.isEmpty(networkStdlib);
-    const hasCustomDeploy = hasNetworkStdlib && networkStdlib.customDeploy;
-    const customDeployMatches = hasCustomDeploy && this.areSameStdlib(networkStdlib, this.packageData.stdlib);
-
+    const customDeployMatches = this.networkFile.hasCustomDeploy() && this.packageFile.hasStdlib(this.networkFile.stdlib)
     if (customDeployMatches) {
-      log.info(`Using existing custom deployment of stdlib at ${networkStdlib.address}`);
-      await this.app.setStdlib(networkStdlib.address);
-      return;
+      log.info(`Using existing custom deployment of stdlib at ${this.networkFile.stdlibAddress}`);
+      return this.app.setStdlib(this.networkFile.stdlibAddress);
```

Comment:
`await` missing?

Reply:
No need given we are returning an async call